python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll

```diff
@@ -29,7 +52,32 @@ module UrlRedirect {
   /**
    * A sanitizer for "URL redirection" vulnerabilities.
    */
-  abstract class Sanitizer extends DataFlow::Node { }
+  abstract class Sanitizer extends DataFlow::Node {
+    /**
+     * Holds if this sanitizer sanitizes flow in the given state.
+     *
+     * By default, sanitizers sanitize all flow, but some sanitiziers, for example,
+     * do not handle untrusted input that contains backslashes, so they only sanitize
+     * flow in the `NoBackslashes` state.
+     */
+    predicate sanitizes(FlowState state) { any() }
```

Comment:
Makes sense!

Reply:
I think I had made it concrete to avoid existing subclasses having to override the newly introduced predicate, but that's perhaps not an overly pressing concern.